node/src/network/computation.rs

```diff
@@ -33,9 +33,7 @@ pub trait MpcLeaderCentricComputation<T>: Sized + 'static {
 
         // We'll wrap the following future in a timeout below.
         let fut = async move {
-            if !sender.is_leader() {
-                sender.wait_for_all_participants_connected().await?;
-            }
+            sender.wait_for_all_participants_connected().await?;
```

Comment:
FYI this isn't necessary for the leader since the way it chooses the participants already means the participants are connected. But it's also fine to make this change.